src/libs/actions/Task.js

```diff
@@ -335,10 +364,32 @@ function reopenTask(taskReportID, taskTitle) {
         },
     ];
 
-    // Update optimistic data for parent report action
+    // Update optimistic data and failure data for parent report action
     const optimisticParentReportData = ReportUtils.getOptimisticDataForParentReportAction(taskReportID, reopenedTaskReportAction.created, CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD);
     if (!_.isEmpty(optimisticParentReportData)) {
         optimisticData.push(optimisticParentReportData);
+        const assigneeChatReportID = lodashGet(ReportUtils.getChatByParticipants(taskReport.participantAccountIDs), 'reportID');
+        if (assigneeChatReportID && `${assigneeChatReportID}` !== `${taskReport.parentReportID}`) {
+            const parentReportAction = ReportActionsUtils.getParentReportActionForTask(taskReportID, assigneeChatReportID);
+            if (parentReportAction) {
+                optimisticData.push({
+                    onyxMethod: Onyx.METHOD.MERGE,
+                    key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${assigneeChatReportID}`,
+                    value: {
+                        [parentReportAction.reportActionID]: _.values(optimisticParentReportData.value)[0],
+                    },
```

Comment:
Can we make this a little more readable? It's not clear what value we are accessing from the array

Reply:
added comment